core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -475,11 +468,20 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
+  private val responseQueue = new LinkedBlockingDeque[RequestChannel.Response]()
+
   private[kafka] val metricTags = mutable.LinkedHashMap(
     "listener" -> listenerName.value,
     "networkProcessor" -> id.toString
   ).asJava
 
+  newGauge("ResponseQueueSize",
```

Comment:
Doesn't this change the metric name?

Reply:
Yeah, you are right. I forgot about the type attribute. @rajinisivaram Maybe you can fix it in #4539?

Reply:
@hachikuji I had forgotten too. Yes, I will update in #4539